src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1514,56 +1514,81 @@ private IOperation CreateBoundLocalDeclarationOperation(BoundLocalDeclaration bo
+                        varStatement = statement;
 
-                    // var declaration points to VariableDeclaratorSyntax
-                    varDeclaration = statement.Declaration.Variables.First();
-                    break;
-                }
+                        varDeclaration = statement.Declaration;
+
+                        varDeclarator = statement.Declaration.Variables.First();
+                        break;
+                    }
                 case SyntaxKind.VariableDeclarator:
-                {
-                    // this happen for 'for loop' initializer
-                    // var statement points to VariableDeclarationSyntax
-                    varStatement = node.Parent;
+                    {
+                        // this happen for 'for loop' initializer
+                        // We generate a DeclarationGroup for this scenario to maintain tree shape consistency across IOperation.
+                        // var statement points to VariableDeclarationSyntax
+                        varStatement = node.Parent;
 
-                    // var declaration points to VariableDeclaratorSyntax
-                    varDeclaration = node;
-                    break;
-                }
+                        varDeclaration = node.Parent;
+
+                        // var declaration points to VariableDeclaratorSyntax
+                        varDeclarator = node;
+                        break;
+                    }
                 default:
-                {
-                    Debug.Fail($"Unexpected syntax: {kind}");
+                    {
+                        Debug.Fail($"Unexpected syntax: {kind}");
 
-                    // otherwise, they points to whatever bound nodes are pointing to.
-                    varStatement = varDeclaration = node;
-                    break;
-                }
+                        // otherwise, they points to whatever bound nodes are pointing to.
+                        varStatement = varDeclaration = varDeclarator = node;
+                        break;
+                    }
             }
 
-            Lazy<ImmutableArray<IVariableDeclarationOperation>> declarations = new Lazy<ImmutableArray<IVariableDeclarationOperation>>(() => ImmutableArray.Create(CreateVariableDeclarationInternal(boundLocalDeclaration, varDeclaration)));
+            Lazy<ImmutableArray<IVariableDeclaratorOperation>> declarations = new Lazy<ImmutableArray<IVariableDeclaratorOperation>>(() => ImmutableArray.Create(CreateVariableDeclarationInternal(boundLocalDeclaration, varDeclarator)));
+            bool multiVariableImplicit = boundLocalDeclaration.WasCompilerGenerated;
+            // In C#, the MultiVariable initializer will always be null, but we can't pass null as the actual lazy. We assume that all lazy elements always exist
+            Lazy<IVariableInitializerOperation> initializer = OperationFactory.EmptyInitializer;
```

Comment:
OperationFactory.EmptyLazy<IVariableInitializerOpertion>()? implementing => new Lazy<T>(null);

Reply:
I filed [https://github.com/dotnet/roslyn/issues/22938](https://github.com/dotnet/roslyn/issues/22938) to address all instances of that.

---
In reply to: [148085638](https://github.com/dotnet/roslyn/pull/22842#discussion_r148085638) [](ancestors = 148085638)